internal/stacks/stackruntime/internal/stackeval/output_value.go

```diff
@@ -146,7 +146,26 @@ func (v *OutputValue) CheckResultValue(ctx context.Context, phase EvalPhase) (ct
 				return cfg.markResultValue(cty.UnknownVal(ty)), diags
 			}
 
-			if !cfg.Declaration(ctx).Ephemeral {
+			if cfg.Declaration(ctx).Ephemeral {
+				// Verify that ephemeral outputs are not declared on the root stack.
+				if v.Addr().Stack.IsRoot() {
+					diags = diags.Append(result.Diagnostic(
+						tfdiags.Error,
+						"Ephemeral output value not allowed on root stack",
+						fmt.Sprintf("Output value %q is marked as ephemeral, this is only allowed in embedded stacks.", v.Addr().Item.Name),
+					))
+				}
+
+				// Verify that the value is ephemeral.
+				if !marks.Contains(result.Value, marks.Ephemeral) {
+					diags = diags.Append(result.Diagnostic(
+						tfdiags.Error,
+						"Expected ephemeral value",
+						fmt.Sprintf("The output value %q is marked as ephemeral, but the value is not ephemeral.", v.Addr().Item.Name),
+					))
+				}
```

Comment:
could we just mark it as ephemeral at this point instead of erroring? I'm not familiar with the expected behaviour, but that is what we do with sensitive.

Reply:
We try to make it difficult the user to use an ephemeral value in a context they don't consider ephemeral so we are a bit more explicit and strict than maybe necessary. It's easier to get rid of restrictions later on if needed.